tests/src/Kernel/AlterableSchemaTest.php

```diff
@@ -0,0 +1,164 @@
+    $result = $this->query('query { alterableQuery { id } }');
+    $this->assertSame(200, $result->getStatusCode());
+    // The should be error that query argument data is required.
+    $this->assertSame([
+      'errors' => [
+        0 => [
+          'message' => 'Field "alterableQuery" argument "id" of type "Int!" is required but not provided.',
+          'extensions' => [
+            'category' => 'graphql',
+          ],
+          'locations' => [
+            0 => [
+              'line' => 1,
+              'column' => 9,
+            ],
+          ],
+        ],
+      ],
+    ], json_decode($result->getContent(), TRUE));
+  }
+
+  /**
+   * Test if schema extension altering is working and arg position is non-null.
+   */
+  public function testSchemaExtensionAlteredQueryResultPropertyToNonNull(): void {
+    $result = $this->query('query { alterableQuery(id: 1) { id, position } }');
+    $this->assertSame(200, $result->getStatusCode());
+    // The should be error that query argument data position is required.
+    $this->assertSame([
+      'errors' => [
+        0 => [
+          'message' => 'Internal server error',
+          'extensions' => [
+            'category' => 'internal',
+          ],
+          'locations' => [
+            0 => [
+              'line' => 1,
+              'column' => 37,
+            ],
+          ],
+          'path' => [
+            'alterableQuery',
+            'position',
+          ],
+        ],
+      ],
+      'data' => [
+        'alterableQuery' => NULL,
+      ],
+    ], json_decode($result->getContent(), TRUE));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function mockSchema($id, $schema, array $extensions = []): void {
+    /** @var \PHPUnit\Framework\MockObject\MockObject $extensionManager */
+    $extensionManager = $this->getMockBuilder(SchemaExtensionPluginManager::class)
+      ->disableOriginalConstructor()
+      ->onlyMethods(['getExtensions'])
+      ->getMock();
+
+    // Adds extra extension in order to test alter extension data event.
+    $extensions['graphql_alterable_schema_test'] = $this->getMockBuilder(SdlSchemaExtensionPluginBase::class)
+      ->disableOriginalConstructor()
+      ->onlyMethods(['getBaseDefinition', 'getExtensionDefinition'])
+      ->getMockForAbstractClass();
+
+    $extensions['graphql_alterable_schema_test']->expects(static::any())
+      ->method('getBaseDefinition')
+      ->willReturn('');
+
+    $extensions['graphql_alterable_schema_test']->expects(static::any())
+      ->method('getExtensionDefinition')
+      ->willReturn(
+        <<<GQL
+          extend type Result {
+            position: Int
+          }
+        GQL
+      );
+
+    $extensionManager->expects(static::any())
+      ->method('getExtensions')
+      ->willReturn($extensions);
+
+    // Replace mock schema with out own implementation.
+    $this->schema = $this->getMockBuilder(AlterableComposableTestSchema::class)
```

Comment:
why do you need to mock this class when it is only a test class that you can fully change in the test?

Reply:
Could you suggest better way? Cause I need to use class where events can be triggered.

Reply:
Hm, you could write your own constructor on AlterableComposableTestSchema and then implement the class methods there to return exactly the things you are mocking here with the method calls.

But should not matter much, I'm also fine to leave it like this.

Reply:
I'll leave it as is :) Thank you for the discussion.